torchvision/transforms/v2/functional/_utils.py

```diff
@@ -6,3 +7,25 @@
 
 def is_simple_tensor(inpt: Any) -> bool:
     return isinstance(inpt, torch.Tensor) and not isinstance(inpt, Datapoint)
+
+
+_KERNEL_REGISTRY = {}
+
+
+def register_kernel(dispatcher, datapoint_cls, *, datapoint_wrapping=True):
+    def datapoint_wrapper(kernel):
+        @functools.wraps(kernel)
+        def wrapper(inpt, *args, **kwargs):
+            return type(inpt).wrap_like(inpt, kernel(inpt.as_subclass(torch.Tensor), *args, **kwargs))
+
+        return wrapper
+
+    def decorator(kernel):
+        _KERNEL_REGISTRY[(dispatcher, datapoint_cls)] = datapoint_wrapper(kernel) if datapoint_wrapping else kernel
+        return kernel
+
+    return decorator
+
+
+def _get_kernel(dispatcher, datapoint_cls):
+    return _KERNEL_REGISTRY[(dispatcher, datapoint_cls)]
```

Comment:
Right now this only supports exact type match, but we can easily extend this to also support subclasses, i.e. `MyImage(datapoints.Image)`.